### PR TITLE
Show loader when rebasing

### PR DESCRIPTION
**Pull request by KarlHeitmann**
- **PR Description**

Solves issue #1488  and also #2610 . Shows WithLoaderPanel window when a rebase simple is make.

I tested it with a big repository I have, and switched rebased from one commit to the other.

If am missing anything just let me know so I can fix it.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/i18n/english.go [hunk 1]
```diff
@@ -795,6 +796,7 @@ func EnglishTranslationSet() TranslationSet {
 		NoChangedFiles:                      "No changed files",
 		PullWait:                            "Pulling...",
 		PushWait:                            "Pushing...",
+		RebaseWait:                          "Rebasing...",
```
Comment (jesseduffield) [inline pkg/i18n/english.go:799]: The fields ending in 'Wait' are for popup panels, but for rebases I don't want a popup. Instead we want to add a new field under the `Actions` field for this and have the value be `Rebasing` (no `...`)

File: pkg/gui/controllers/helpers/merge_and_rebase_helper.go [hunk 2]
```diff
@@ -224,8 +225,10 @@ func (self *MergeAndRebaseHelper) RebaseOntoRef(ref string) error {
 			Key:   's',
 			OnPress: func() error {
 				self.c.LogAction(self.c.Tr.Actions.RebaseBranch)
-				err := self.c.Git().Rebase.RebaseBranch(ref)
-				return self.CheckMergeOrRebase(err)
+				return self.c.WithLoaderPanel(self.c.Tr.RebaseWait, func(task gocui.Task) error {
```
Comment (jesseduffield) [inline pkg/gui/controllers/helpers/merge_and_rebase_helper.go:228]: Let's use `WithWaitingStatus` instead as that's the pattern we use for things that don't involve talking to a remote

**Review by jesseduffield — COMMENTED** [review]
Couple comments :)

File: pkg/gui/controllers/helpers/merge_and_rebase_helper.go (hunk 2 above)
Comment (KarlHeitmann) [reply pkg/gui/controllers/helpers/merge_and_rebase_helper.go]: I think I've read in one of your comments in other issue to use something different that `WithWaitingStatus` , but I was lazy to search it again.  Now I have changed it, it is a spinner at the left bottom corner of the UI.

**Comment by KarlHeitmann** [comment]
Submitted changes, please let me know if there is something more I can do there.

File: pkg/i18n/english.go [hunk 3]
```diff
@@ -1323,6 +1324,7 @@ func EnglishTranslationSet() TranslationSet {
 			ForceCheckoutBranch:               "Force checkout branch",
 			DeleteBranch:                      "Delete branch",
 			Merge:                             "Merge",
+			Rebase:                            "Rebasing",
```
Comment (jesseduffield) [inline pkg/i18n/english.go:1327]: My apologies: looks like Actions is the wrong place to put this (it's using present tense and everything else is tenseless). In fact, we already have a `RebasingStatus` to use! So let's just use the existing one

**Review by jesseduffield — CHANGES_REQUESTED** [review]
One more thing

**Comment by KarlHeitmann** [comment]
> One more thing

I think I've got it now @jesseduffield , I rebased the branch, solved the conflicts one by one and verified in each commit the spinner appeared. I don't know if you are going to squash the changes later or not, but that's what I did.

Let me know if there are more things to fix or if there is something else missing.

Best

**Comment by jesseduffield** [comment]
LGTM. Can you squash the commits?

**Comment by KarlHeitmann** [comment]
Ready @jesseduffield !

**Comment by jesseduffield** [comment]
Great work @KarlHeitmann

**Pull request merged**